src/payment/strategies/paypal-commerce/paypal-commerce-sdk.ts

```diff
@@ -54,10 +54,18 @@ export interface PaypalCommerceHostWindow extends Window {
     paypal?: PaypalCommerceSDK;
 }
 
+export interface PaypalCommerceInitializationData {
+    clientId: string;
+    intent?: 'capture' | 'authorize';
+    isPayPalCreditAvailable?: boolean;
+}
+
+const ScriptDisableFunding = [ 'credit', 'card' ];
+
 export interface PaypalCommerceScriptOptions {
     clientId: string;
     currency?: string;
     commit?: boolean;
     intent?: 'capture' | 'authorize';
-    disableFunding?: 'credit';
+    disableFunding?: Array<typeof ScriptDisableFunding[number]>;
```

Comment:
A simpler way to declare this property is `Array<'credit' | 'card'>`.

Reply:
Thank you!
But in this case I see error "Type 'string' is not assignable to type 'credit' | 'card''." I can rewrite some code for this, but it's not so elegant :slightly_smiling_face:
So I think it would be better use `Array<typeof ScriptDisableFunding[number]>`

Reply:
From what I can tell, you're not seeing the `'string' is not assignable` error because `Array<typeof ScriptDisableFunding[number]>` essentially means `string[]`. This is because with this line 

```ts
const ScriptDisableFunding = [ 'credit', 'card' ]
```

you're declaring a variable `ScriptDisableFunding` with `string[]` as its type (it is implicitly declared by TS). Therefore `Array<typeof ScriptDisableFunding[number]>` translates to `Array<string[][number]>`, which is simply `string[]`.

I think what you're trying to do here is to declare this property to only accept an array containing either `'credit'` or `'card'`, not any string. If you want to achieve that, `Array<'credit' | 'card'>` is the correct declaration. Therefore you should look at fixing the line that gives you the error instead. For example, in this file `paypal-commerce-button-strategy.ts`, you have

```ts
const disableFunding = [ 'card' ];
```

This is similar to the issue I described above where the type for that variable is implicitly declared as `string[]`. So when you assign it to a property that has a stricter type, you'll get a type error. Instead, what you need to do is to either provide a type for that variable explicitly, i.e.: 

```ts
const disableFunding: Array<'credit' | 'card'>
```
or inline the declaration with the return value: i.e.:

```ts
return {
    clientId,
    commit: false,
    currency: cart.currency.code,
    disableFunding: isPayPalCreditAvailable ? ['card', 'credit'] : ['card'],
    intent,
};
```

Reply:
You are right! Thank you!

I created type disableFundingType for avoidance duplicate code. Because here there are can be more options (you can see it here https://developer.paypal.com/docs/checkout/reference/customize-sdk/#disable-funding)